### PR TITLE
[FASTFAT_NEW] Fix MsVC builds

### PR DESCRIPTION
**Pull request by SergeGautherie**
## Purpose

JIRA issue: [CORE-14879](https://jira.reactos.org/browse/CORE-14879)

### CODE REVIEW

File: drivers/filesystems/fastfat_new/fatinit.c [hunk 1]
```diff
@@ -15,6 +15,10 @@ Module Name:
 
 #include "fatprocs.h"
 
+#ifdef __REACTOS__
+    #define _Unreferenced_parameter_
```
Comment (HeisSpiter) [inline drivers/filesystems/fastfat_new/fatinit.c]: Fix indent

File: drivers/filesystems/fastfat_new/fatprocs.h [hunk 2]
```diff
@@ -39,7 +39,7 @@ Module Name:
 #endif
 
 
-#ifdef __REACTOS__
+#ifdef __REACTOS__  // Copied from below.
```
Comment (HeisSpiter) [inline drivers/filesystems/fastfat_new/fatprocs.h]: Useless change, revert

File: drivers/filesystems/fastfat_new/fatprocs.h [hunk 3]
```diff
@@ -1026,7 +1026,7 @@ FatUnpinEaRange (
 //  of the file object, implemented in FilObSup.c
 //
 
-#ifndef __REACTOS__
+#ifndef __REACTOS__  // Copied to above.
```
Comment (HeisSpiter) [inline drivers/filesystems/fastfat_new/fatprocs.h]: Useless change, revert

File: drivers/filesystems/fastfat_new/fatstruc.h [hunk 4]
```diff
@@ -19,6 +19,9 @@ Module Name:
 
 typedef PVOID PBCB;     //**** Bcb's are now part of the cache module
 
+#ifdef __REACTOS__
+    #define __volatile volatile
```
Comment (HeisSpiter) [inline drivers/filesystems/fastfat_new/fatstruc.h]: Fix indent

File: drivers/filesystems/fastfat_new/fatprocs.h (hunk 2 above)
Comment (SergeGautherie) [reply drivers/filesystems/fastfat_new/fatprocs.h]: Reverted.

File: drivers/filesystems/fastfat_new/fatprocs.h (hunk 3 above)
Comment (SergeGautherie) [reply drivers/filesystems/fastfat_new/fatprocs.h]: Reverted.

File: drivers/filesystems/fastfat_new/fatinit.c (hunk 1 above)
Comment (SergeGautherie) [reply drivers/filesystems/fastfat_new/fatinit.c]: Removed.

File: drivers/filesystems/fastfat_new/fatstruc.h (hunk 4 above)
Comment (SergeGautherie) [reply drivers/filesystems/fastfat_new/fatstruc.h]: Removed.

**Comment by HBelusca** [comment]
Could use the `RTL_MAKE_LARGE_INTEGER(low_part, high_part)` ReactOS macro too.

**Comment by SergeGautherie** [comment]
> `RTL_MAKE_LARGE_INTEGER()`

Probably, and/or `RTL_CONSTANT_LARGE_INTEGER()`.
Nonetheless, these macros are not/rarely used, and I assume minimal diff is preferred.

**Comment by SergeGautherie** [comment]
> ReactOS macro

Also, shouldn't these 2 macros be inside `#ifdef __REACTOS__`?

File: drivers/filesystems/fastfat_new/fatinit.c [hunk 5]
```diff
@@ -15,6 +15,10 @@ Module Name:
 
 #include "fatprocs.h"
 
+#ifdef __REACTOS__
```
Comment (HBelusca) [inline drivers/filesystems/fastfat_new/fatinit.c]: Why not grouping _Unreferenced_parameter_ and __volatile defines together in a single .h file (for example in the precompiled header) ?

Comment (SergeGautherie) [reply drivers/filesystems/fastfat_new/fatinit.c]: I started with a local change, as this define is used there only.
Nonetheless, I could move the latter, if requested.

**Pull request merged**